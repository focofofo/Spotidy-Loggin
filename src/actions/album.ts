import { AlbumPayload } from '../models'

export const SET_ALBUMS = 'SET_ALBUMS'
export const LOAD_MORE_ALBUMS = 'LOAD_MORE_ALBUMS'
export const RESET_ALBUMS = 'RESET_ALBUMS'

/**
 * Set the albums information
 * @param payload array with albums
 */
export function doSetAlbums(payload: AlbumPayload) {
  return {
    type: SET_ALBUMS,
    payload,
  }
}

/**
 * Add the new array of albums to the current state
 * @param payload array with new albums
 */
export function doLoadMoreAlbums(payload: AlbumPayload) {
  return {
    type: LOAD_MORE_ALBUMS,
    payload,
  }
}

/**
 * Clear the album information
 */
export function doResetAlbums() {
  return { type: RESET_ALBUMS }
}
