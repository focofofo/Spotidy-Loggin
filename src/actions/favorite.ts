import { FavoritePayload } from '../models'

export const SET_FAVORITES = 'SET_FAVORITES'
export const REMOVE_FAVORITE = 'REMOVE_FAVORITE'
export const RESET_FAVORITES = 'RESET_FAVORITES'

/**
 * Save the favorite list into the storage
 * @param payload favorite list result of the API
 */
export function doSetFavorites(payload: FavoritePayload) {
  return {
    type: SET_FAVORITES,
    payload,
  }
}

/**
 * Removes the favorite track from the storage given the track index
 * @param payload index of the track to be removed
 */
export function doRemoveFavorite(payload: number) {
  return {
    type: REMOVE_FAVORITE,
    payload,
  }
}

/**
 * Clear the favorite storage
 */
export function doResetFavorites() {
  return {
    type: RESET_FAVORITES,
  }
}
