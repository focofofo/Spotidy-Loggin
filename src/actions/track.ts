import { TrackPayload } from '../models'

export const SET_TRACKS = 'SET_TRACKS'
export const ALBUM_NOT_FOUND = 'ALBUM_NOT_FOUND'
export const ADD_FAVORITE_TRACK = 'ADD_FAVORITE_TRACK'
export const REMOVE_FAVORITE_TRACK = 'REMOVE_FAVORITE_TRACK'
export const CHECK_FAVORITES = 'CHECK_FAVORITES'
export const RESET_TRACKS = 'RESET_TRACKS'

/**
 * Set the album's tracks information into the storage
 * @param payload album's tracks information from the API
 */
export function doSetTracks(payload: TrackPayload) {
  return {
    type: SET_TRACKS,
    payload,
  }
}

/**
 * Indicates the error in case the album is not found
 */
export function doAlbumNotFound() {
  return {
    type: ALBUM_NOT_FOUND,
  }
}

/**
 * Update the is_favorite key of the specified track
 * @param payload index from the tracks storage
 */
export function doAddFavorite(payload: number) {
  return {
    type: ADD_FAVORITE_TRACK,
    payload,
  }
}

/**
 * Remove the is_favorite key of the specified track
 * @param payload index from the tracks storage
 */
export function doRemoveFavoriteTrack(payload: number) {
  return {
    type: REMOVE_FAVORITE_TRACK,
    payload,
  }
}

/**
 * Send to storage the liked songs information
 * @param payload array of booleans indicating if track is liked or not
 */
export function doCheckFavorites(payload: boolean[]) {
  return {
    type: CHECK_FAVORITES,
    payload,
  }
}

/**
 * Clear the tracks storage
 */
export function doResetTracks() {
  return {
    type: RESET_TRACKS,
  }
}
