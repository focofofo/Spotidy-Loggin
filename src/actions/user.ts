import { UserInformationPayload } from '../models'

export const SET_ACCESS_TOKEN = 'SET_ACCESS_TOKEN'
export const SET_USER_INFORMATION = 'SET_USER_INFORMATION'
export const LOGOUT_USER = 'LOGOUT_USER'

/**
 * Set access_token to the user's state
 * @param payload access_token for the user
 */
export function doSetAccessToken(payload: string) {
  return {
    type: SET_ACCESS_TOKEN,
    payload,
  }
}

/**
 * Set the user information from API response
 * @param payload id, display_name from the user
 */
export function doSetUserInformation(payload: UserInformationPayload) {
  return {
    type: SET_USER_INFORMATION,
    payload,
  }
}

/**
 * Log out the current user
 */
export function doLogoutUser() {
  return { type: LOGOUT_USER }
}
