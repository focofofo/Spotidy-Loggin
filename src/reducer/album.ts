import { AnyAction } from 'redux'
import { AlbumPayload } from '../models'
import { SET_ALBUMS, RESET_ALBUMS, LOAD_MORE_ALBUMS } from '../actions/album'

const initialState: AlbumPayload = {
  href: '',
  items: [],
  limit: 0,
  next: '',
  offset: 0,
  previous: '',
  total: 0,
}

export default function album(
  state: AlbumPayload = initialState,
  action: AnyAction
) {
  switch (action.type) {
    case SET_ALBUMS:
      return {
        ...state,
        ...action.payload,
      }
    case LOAD_MORE_ALBUMS:
      return {
        ...state,
        items: [...state.items, ...action.payload.items],
        next: action.payload.next,
      }
    case RESET_ALBUMS:
      return { ...initialState }
    default:
      return state
  }
}
