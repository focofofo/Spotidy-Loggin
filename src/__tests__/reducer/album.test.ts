import { AnyAction } from 'redux'
import album from '../../reducer/album'
import { SET_ALBUMS, LOAD_MORE_ALBUMS, RESET_ALBUMS } from '../../actions/album'

describe('album reducer', () => {
  it('should handle initial state', () => {
    expect(album(undefined, {} as AnyAction)).toMatchSnapshot()
  })

  it('should handle SET_ALBUMS', () => {
    expect(
      album(undefined, {
        type: SET_ALBUMS,
        payload: {
          href: 'test',
          items: ['test'],
          limit: 10,
          next: 'test',
          offset: 10,
          previous: 'test',
          total: 10,
        },
      })
    ).toMatchSnapshot()
  })

  it('should handle LOAD_MORE_ALBUMS', () => {
    expect(
      album(undefined, {
        type: LOAD_MORE_ALBUMS,
        payload: {
          href: 'test',
          items: ['test'],
          limit: 10,
          next: 'test',
          offset: 10,
          previous: 'test',
          total: 10,
        },
      })
    ).toMatchSnapshot()
  })

  it('should handle RESET_ALBUMS', () => {
    expect(album(undefined, { type: RESET_ALBUMS })).toMatchSnapshot()
  })
})
