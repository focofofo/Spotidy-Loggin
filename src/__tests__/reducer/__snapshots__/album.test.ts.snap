// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`album reducer should handle LOAD_MORE_ALBUMS 1`] = `
Object {
  "href": "",
  "items": Array [
    "test",
  ],
  "limit": 0,
  "next": "test",
  "offset": 0,
  "previous": "",
  "total": 0,
}
`;

exports[`album reducer should handle RESET_ALBUMS 1`] = `
Object {
  "href": "",
  "items": Array [],
  "limit": 0,
  "next": "",
  "offset": 0,
  "previous": "",
  "total": 0,
}
`;

exports[`album reducer should handle SET_ALBUMS 1`] = `
Object {
  "href": "test",
  "items": Array [
    "test",
  ],
  "limit": 10,
  "next": "test",
  "offset": 10,
  "previous": "test",
  "total": 10,
}
`;

exports[`album reducer should handle initial state 1`] = `
Object {
  "href": "",
  "items": Array [],
  "limit": 0,
  "next": "",
  "offset": 0,
  "previous": "",
  "total": 0,
}
`;
